AllReadyApp/Web-App/AllReady/Areas/Admin/Controllers/InviteController.cs

```diff
@@ -0,0 +1,154 @@
+
+namespace AllReady.Areas.Admin.Controllers
+{
+    [Area("Admin")]
+    [Authorize(nameof(UserType.OrgAdmin))]
+    public class InviteController : Controller
+    {
+        private readonly IMediator _mediator;
+        private readonly UserManager<ApplicationUser> _userManager;
+
+        public InviteController(IMediator mediator, UserManager<ApplicationUser> userManager)
+        {
+            _mediator = mediator;
+            _userManager = userManager;
+        }
+
+        // GET: Admin/Invite/SendCampaignManagerInvite/5
+        [Route("[area]/[controller]/[action]/{campaignId:int}")]
+        public async Task<IActionResult> SendCampaignManagerInvite(int campaignId)
+        {
+            var campaign = await _mediator.SendAsync(new CampaignByCampaignIdQuery { CampaignId = campaignId });
+            if (campaign == null) return NotFound();
+
+            if (!IsUserAuthorizedToSendInvite(campaign.ManagingOrganizationId))
+            {
+                return Unauthorized();
+            }
+
+            return View("Send", new InviteViewModel
+            {
+                InviteType = InviteType.CampaignManagerInvite,
+                CampaignId = campaignId,
+            });
+        }
+
+        // GET: Admin/Invite/SendEventManagerInvite/5
+        [Route("[area]/[controller]/[action]/{eventId:int}")]
+        public async Task<IActionResult> SendEventManagerInvite(int eventId)
+        {
+            var @event = await _mediator.SendAsync(new EventByEventIdQuery { EventId = eventId });
+            if (@event == null) return NotFound();
+
+            if (!IsUserAuthorizedToSendInvite(@event.Campaign.ManagingOrganizationId))
+            {
+                return Unauthorized();
+            }
+
+            return View("Send", new InviteViewModel
+            {
+                InviteType = InviteType.EventManagerInvite,
+                EventId = eventId,
+            });
+        }
+
+        // POST: Admin/Invite/SendCampaignManagerInvite/5
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        [Route("[area]/[controller]/[action]/{campaignId:int}")]
+        public async Task<IActionResult> SendCampaignManagerInvite(int campaignId, InviteViewModel invite)
+        {
+            if (invite == null)
+            {
+                return BadRequest();
+            }
+
+            if (ModelState.IsValid)
+            {
+                var campaign = await _mediator.SendAsync(new CampaignByCampaignIdQuery { CampaignId = invite.CampaignId });
+
+                if (campaign == null)
+                {
+                    return BadRequest();
+                }
+
+                if (!IsUserAuthorizedToSendInvite(campaign.ManagingOrganizationId))
+                {
+                    return Unauthorized();
+                }
+
+                var user = await _userManager.GetUserAsync(User);
+                await _mediator.SendAsync(new CreateInviteCommand { Invite = invite, UserId = user?.Id });
```

Comment:
You should break down the invite for the approapriate type into separate command and separate handlers... for instance, instead of having `CreateInviteCommand`, break the work up into smaller logical units with a `CreateManagerInviteCommand` and `CreateEventInviteCommand`. The same goes for the handlers of those commands. the handlers should be very simple, we don't want too much logic in them.